src/lxc/conf.c

```diff
@@ -1433,6 +1434,40 @@ static int lxc_setup_devpts(int num_pts)
 		DEBUG("removed existing \"/dev/ptmx\"");
 	}
 
+	/* Read attributes of /dev/pts/ptmx */
+	ret = stat("/dev/pts/ptmx", &ptmx);
+	if (ret) {
+		SYSERROR("failed to stat \"/dev/pts/ptmx\"");
+		return -1;
+	} else if (!S_ISCHR(ptmx.st_mode)) {
```

Comment:
Since you return in the  `if` branch above the `else if` is not really needed. Avoid the indentation and just perform the check unconditionally. Meaning:

```
if (/* char dev check fails */)
       return error;
```